modules/cuml/src/utilities/array_utils.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  DataFrame,
+  Int32,
+  Numeric,
+  Series,
+  TypeMap,
+} from '@rapidsai/cudf';
+import {DeviceBuffer} from '@rapidsai/rmm/';
+
+/**
+ * convert a dataframe to a single series to replicate conversion to a matrix in an organized
+ * for {x1,y1,x2,y2...} for df = {x:[x1,x2], y: [y1,y2]}
+ */
+export function dataframe_to_series(input: DataFrame): Series<Numeric> {
+  return input.interleaveColumns();
+}
+
+/**
+ * convert input of type Series or DataFrame to a DeviceBuffer
+ * @param input
+ * @returns
+ */
+export function transform_input_to_device_buffer(input: Series|DataFrame<TypeMap>): DeviceBuffer {
+  if (input instanceof DataFrame) { return new DeviceBuffer(dataframe_to_series(input).data); }
+  return new DeviceBuffer(input.data);
+}
```

Comment:
Splitting out the above should allow us to remove this. Not all Series specializations have `data` getters. For those that do, we can use them directly instead of making copies via `series.data.buffer`.